pkg/ccl/backupccl/testdata/backup-restore/restore-regionless

```diff
@@ -0,0 +1,334 @@
+# disabled to run within tenant because multiregion primitives are not supported within tenant
+subtest mrbackup
+new-cluster name=s1 allow-implicit-access disable-tenant localities=us-east-1,us-west-1,eu-central-1
+----
+
+exec-sql
+CREATE DATABASE d PRIMARY REGION "us-east-1" REGIONS "us-west-1", "eu-central-1";
+CREATE TABLE d.t (x INT);
+INSERT INTO d.t VALUES (1), (2), (3);
+----
+
+query-sql
+SELECT region FROM [SHOW REGIONS FROM DATABASE d] ORDER BY 1;
+----
+eu-central-1
+us-east-1
+us-west-1
+
+# backup a MR database and table
+exec-sql
+BACKUP DATABASE d INTO 'nodelocal://1/database_backup/';
+----
+
+exec-sql
+BACKUP TABLE d.t INTO 'nodelocal://1/table_backup/';
+----
+
+subtest end
+
+# create regionless cluster
+subtest restore_regionless_on_regionless
+new-cluster name=s2 share-io-dir=s1 allow-implicit-access disable-tenant
```

Comment:
could add another case that runs a cluster RESTORE with strip_localities into a regionless cluster? I'm sorry our test infra leads to so much copy pasta :(